pkg/codegen/convert/mapper.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convert
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+type Mapper interface {
+	GetMapping(provider string) ([]byte, error)
+}
+
+type pluginMapper struct {
+	entries map[string][]byte
+}
+
+func NewPluginMapper(host plugin.Host, key string, mappings []string) (Mapper, error) {
+	entries := map[string][]byte{}
+
+	// Enumerate _all_ our installed plugins to ask for any mappings they provider. This allows users to
+	// convert aws terraform code for example by just having 'pulumi-aws' plugin locally, without needing to
+	// specify it anywhere on the command line, and without tf2pulumi needing to know about every possible plugin.
+	plugins, err := workspace.GetPlugins()
+	if err != nil {
+		return nil, fmt.Errorf("could not get plugins: %w", err)
+	}
+	// We only care about the latest version of each plugin
```

Comment:
This doesn't make sense to me. I think we should prioritize the latest version of a plugin, but fall back to earlier versions. 

This scenario will come up when plugins remove mappings.

Reply:
That's fair, I hadn't considered the situation that v2.GetMapping returned nil, but v1.GetMapping returned data. That does seem reasonable to support.